src/jit/codegenxarch.cpp

```diff
@@ -6252,6 +6252,51 @@ void CodeGen::genCallInstruction(GenTreePtr node)
 
     if (target != nullptr)
     {
+#ifdef _TARGET_X86_
+        if (((call->gtFlags & GTF_CALL_VIRT_KIND_MASK) == GTF_CALL_VIRT_STUB) && (call->gtCallType == CT_INDIRECT))
+        {
+            // On x86, we need to generate a very specific pattern for indirect VSD calls:
+            //
+            //    3-byte nop
+            //    call dword ptr [eax]
+            //
+            // Where EAX is also used as an argument to the stub dispatch helper. Make
+            // sure that the call target address is computed into EAX in this case.
+
+            assert(target->isContainedIndir());
+
+            // Disable random NOP emission
+            getEmitter()->emitDisableRandomNops();
+
+            GenTreeIndir* indir = target->AsIndir();
+            assert(indir->Addr() == indir->Base());
+            assert(indir->HasBase());
+            assert(!indir->HasIndex());
+            assert(indir->Scale() == 1);
+            assert(indir->Offset() == 0);
+
+            GenTree* base = indir->Base();
+            genConsumeReg(base);
```

Comment:
Consider the case where the address is a `GT_LEA`: in that case, `genConsumeAddress` will call `genConsumeOperands` on the `GT_LEA` node. These nodes will already have been consumed, however, by `genCodeForTreeNode` when it processed the `GT_LEA` node. Calling `genConsumeAddress` would cause a bug in that case.

The only assumption that codegen is making is that the address is not contained (I have local changes that add an assert to that effect). Calling `genConsumeReg` is the right thing here.